syntax = "proto3";

package tutorial;

message Person {
    string name = 1;
    int32 id = 2;
    string email = 3;

    message PhoneNumber {
        optional string number = 1;
        optional PhoneType type = 2;
    }
    
    repeated PhoneNumber phones = 4;
}

enum PhoneType {
    PHONE_TYPE_UNSPECIFIED = 0;
    PHONE_TYPE_MOBILE = 1;
    PHONE_TYPE_HOME = 2;
    PHONE_TYPE_WORK = 3;
}

message AddressBook {
    repeated Person people = 1;
}